src/bin/proto2xml.rs

```diff
@@ -0,0 +1,91 @@
+use clap::Parser;
+use rust_q_sim::io::proto_events::EventsReader;
+use rust_q_sim::io::xml_events::XmlEventsWriter;
+use rust_q_sim::mpi::events::proto::Event;
+use rust_q_sim::mpi::events::EventsPublisher;
+use std::io::{Read, Seek};
+use std::path::PathBuf;
+
+struct StatefullReader<R: Read + Seek> {
+    reader: EventsReader<R>,
+    curr_time_step: (u32, Vec<Event>),
+}
+
+impl<R: Read + Seek> StatefullReader<R> {
+    pub fn load_next(&mut self) -> Option<()> {
+        match self.reader.next() {
+            None => None,
+            Some(time_step) => {
+                //println!("load_next {}: {time_step:?}", self.id);
+                self.curr_time_step = time_step;
+                Some(())
+            }
+        }
+    }
+}
+
+fn main() {
+    let args = InputArgs::parse();
+
+    println!("Proto2Xml with args: {args:?}");
+
+    let mut readers = Vec::new();
+    println!("Reading from Files: ");
+    for i in 0..args.num_parts {
+        let file_string = format!("{}.{i}.pbf", args.path);
+        println!("\t {}", file_string);
+        let file_path = PathBuf::from(file_string);
+        let reader = EventsReader::from_file(&file_path);
+        let wrapper = StatefullReader {
+            reader,
+            curr_time_step: (0, Vec::new()),
+        };
+        readers.push(wrapper);
+    }
+    let output_file_string = format!("{}.xml", args.path);
+    let output_file_path = PathBuf::from(output_file_string);
+    let mut publisher = EventsPublisher::new();
+    publisher.add_subscriber(Box::new(XmlEventsWriter::new(&output_file_path)));
+
+    while !readers.is_empty() {
+        readers.sort_by(|a, b| a.curr_time_step.0.cmp(&b.curr_time_step.0));
```

Comment:
I'm not sure whether this is the fastest implementation. Maybe one can use Rust's BinaryHeap (or we find a faster implementation of a priority queue).

Reply:
Yes, I thought so too while writing this. The bottleneck of the entire thing is probably writing the xml file though. I would rather fix this if it becomes a problem. 

I guess this can be sped up too by splitting reading and writing into separate threads.